runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/observers/DirectStreamObserver.java

```diff
@@ -139,4 +166,29 @@ public void onCompleted() {
       outboundObserver.onCompleted();
```

Comment:
should we check if phaser has been terminated beneath the lock here and if so call onError?  that seems like it would enforce that if send didn't send something due to phaser cancellation that we wouldn't complete successfully.

Reply:
When I look at how this is used in StreamObserverFactory we call phaser.forceTermination in `ForwardingClientREsponseObserver#onDone` so we would always be calling onError here since the onDoneHandler runs and terminates the StreamObserver?